Sources/MongoSwift/BSON/BSONValue.swift

```diff
@@ -512,6 +654,15 @@ extension Double: BSONValue {
         }
     }
 
+    /// Initialize this `Double` from another `BSONNumber`.
+    /// If the passed in number cannot be represented exactly as an `Double`, this initializer will return nil.
+    public init?(fromNumber num: BSONNumber) {
```

Comment:
Note, I couldn't use `init?(exactly num: BSONNumber)` since it conflicts with the existing `init?(exactly)` as part of `BinaryInteger`. Also `init?(from)` conflicts with the `init(from)` in `CodableNumbers`.

Reply:
I know I'm the one who proposed some sort of an initializer accepting a `BSONNumber`, but do you think these actually add much value to the API? 
if it could be named `exactly` it might since people are already used to that... but I'm not sure if there's an occasion you'd want to do this rather than just call `toDouble`?

Reply:
I guess it just exists as an alternative way of calling `toDouble`. In the driver, I don't use them anywhere because, as you said, wherever you can use them you could just use the call it's wrapping instead.

I would say that they can just be removed. Let me know if you think they should stay.

As a side note: the reasoning the initializers weren't added to the protocol was that if something conforms to `BSONNumber`, it is likely that it will also conform to at least one of the many `Numeric` descendant protocols, meaning it would probably have an initializer that takes in the result of one of the `toX` converters. So an `init(BSONNumber)` felt redundant. (@mbroadst I know I said friday I'd have a better answer to this question)

Reply:
I'm good with removing them, sorry I sent you down the path of adding of them.

Reply:
No worries at all, it was definitely a path worth exploring.